theano/sandbox/cuda/basic_ops.py

```diff
@@ -3044,6 +3044,28 @@ def make_node(self, *axis_and_tensors):
         as_tensor_variable_args = [as_cuda_ndarray_variable(x)
                                    for x in tensors]
 
+        # Get joining axis as int
+        axis_int = 0
+        if not isinstance(axis, int):
+            try:
+                # Note : `get_scalar_constant_value` returns a ndarray not
+                # an int
+                axis_int = int(tensor.get_scalar_constant_value(axis))
+
+            except tensor.basic.NotScalarConstantError:
+                pass
+	else:
+	    axis_int = axis
+
+        if (axis_int < 0):
+            # Find tensor with smallest number of dimensions
+            min_dim = -1
```

Comment:
Hmm. I don't know if different number of dimensions are supported, but they certaintly should be in the future.

I think as a rule-of-thumb we should always write the most general code, especially if it's in the "make_node" step which is not going to be the bottleneck anyway.